modelci/app/v1/endpoints/model.py

```diff
@@ -6,25 +6,29 @@
 Date: 6/20/2020
 """
 import asyncio
+import json
 import shutil
 from pathlib import Path
 from typing import List
 
 from fastapi import APIRouter, File, UploadFile, Depends
+from fastapi.exceptions import RequestValidationError
+from pydantic.error_wrappers import ErrorWrapper
+from starlette.responses import JSONResponse
 
 from modelci.hub.manager import register_model
 from modelci.persistence.service import ModelService
+from modelci.persistence.service_ import get_by_id
 from modelci.types.bo import Framework, Engine, Task
-from modelci.types.models import MLModelIn
-from modelci.types.models.mlmodel import MLModelInForm
-from modelci.types.vo.model_vo import ModelDetailOut, ModelListOut, Framework as Framework_, Engine as Engine_, \
+from modelci.types.models import MLModelIn, MLModelInForm
```

Comment:
what are the differences between MLModelIn and MLModelInForm?

shall we use more precise name?

Reply:
MLModelIn: definition to register an ML model into database
MLModelInForm: definition to accept requests for model registration in formdata format. 

The detailed reason for using `MLModelInForm` is the `MLModelIn` only works for `JSON` format request. However, a `formdata` is required as we are uploading a file with the request.

Reply:
The idea is from https://fastapi.tiangolo.com/tutorial/extra-models/?h=userin#multiple-models

Reply:
so can we replace `MLModelIn` with `MLModelJSON`?

Reply:
Not exactly. MLModelIn is used for `model_server.save()`. Its similar data name from `FastAPI` is `UserIn`.

See https://fastapi.tiangolo.com/tutorial/extra-models/?h=userin#multiple-models UserIn.

Reply:
I see. However,

On the one hand, we should learn from the tutorial.

On the other hand, we should consider the situation of our project and make it clear and maintainable.

I believe the best name is the easiest to understand the name. I am so confused about these names....

How do you think of this？Let's discuss @huangyz0918 @univerone

Reply:
`from modelci.types.models import MLModelIn, MLModelInYaml, MLModel` 

this makes me crazy.........

Reply:
if the variable is a model data, it should be end with xxx_data, if it is a model object, end it wth xxx_model. `In` is confused for developers.

Reply:
Will this looks good:

`MLModelPost`, `MLModelPostFormDara` and `MLModelPostYamlData`. The hierarchy of them is:
`MLModelInput`
  |-`MLModelInoutFormData`
  |-`MLModelInputYamlData`

Reply:
Maybe we could implement `from_formdata`, `from_yaml` method under MLModelIn class, and only keep this class to create MLModel?

Reply:
Because these data class are parameters for `publish` API:
![image](https://user-images.githubusercontent.com/36268431/110811939-228f1200-82c2-11eb-8a13-a97ed6f4f5bb.png)
And such design is to make the API handler clean without much data validation.

If we use only one single model, the model is super big and very hard to read.

Reply:
And the inputs fields are quite different.

Reply:
> And the inputs fields are quite different.

I got your point, what if we post json data instead of form data,  could this save a lot of effort?

Reply:
If you are posting a file, you have to use `multipart/formdata` as the header instead of `application/json`. This is required by HTTP.

Anyway, I am trying hard to merge those data class while keeping them simple

Reply:
Now only have `BaseMLModel` (abstraction contains very common attributes), `MLModel` (for show the model and save the model), `MLModelFromYaml` (for save the model from a YAML file)